generate-component/index.ts

```diff
@@ -0,0 +1,98 @@
+import fsSync, { promises as fs } from "fs";
+import { parseArgs } from "node:util";
+import { component, story, test } from "./component_templates";
+
+function writeFileErrorHandler(err) {
+  if (err) throw err;
+}
+
+async function updateIndexFile(newTemplateName) {
+  const indexFileContents = await fs.readFile(
+    "./packages/syntax-core/src/index.tsx",
+    "utf8",
+  );
+
+  // grab all components and combine them with new component
+  const currentComponents =
+    indexFileContents.match(/(?<=import )(.*?)(?= from)/g) || "";
+  const newComponents = [newTemplateName, ...currentComponents].sort();
+
+  // create the import and export statements
+  const importStatements = newComponents
+    .map(
+      (importName) =>
+        `import ${importName} from './${importName}/${importName}';\n`,
+    )
+    .join("");
+  const exportStatements = `export {\n${newComponents
+    .map((component) => `  ${component},\n`)
+    .join("")}};\n`;
+
+  const fileContent = `${importStatements}\n${exportStatements}`;
+
+  try {
+    await fs.writeFile(`./packages/syntax-core/src/index.tsx`, fileContent);
+  } catch (error) {
+    writeFileErrorHandler(error);
+  }
+}
+
+async function generateComponent() {
+  // grab component name from terminal argument
+  const { positionals } = parseArgs({
+    options: {},
+    allowPositionals: true,
+  });
+  const name = positionals[0];
+  if (!name) throw new Error("You must include a component name.");
+
+  const dir = `./packages/syntax-core/src/${name}/`;
+
+  // Throw an error if the file already exists
+  if (fsSync.existsSync(dir))
+    throw new Error("A component with that name already exists.");
+
+  try {
```

Comment:
I wasn't sure if each `read + writeFile` should be it's own `try catch` @christianvuerings do you have opinions on this? ideally we want it to fail if any of these fail,  so I think this is fine